backend/tests/test_media_handler.py

```diff
@@ -0,0 +1,93 @@
+import tempfile
+import unittest
+from unittest.mock import patch, MagicMock
+from pathlib import Path
+from backend.media_handler import MediaHandler
+
+class TestMediaHandler(unittest.TestCase):
+
+    def setUp(self):
+        """Set up base directory and mock paths for testing."""
+        self.base_directory = tempfile.mkdtemp()
+        self.video_url = "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
+        self.config_path = "/path/to/config.json"
+        self.video_path = "/path/to/video.mp4"
+        self.input_data = {
+            "video_file_path": self.video_path,
+            "config_file_path": self.config_path
+        }
+
+    @patch("backend.media_handler.yt_dlp.YoutubeDL")
+    def test_download_media(self, mock_yt_dlp):
+        # Mock YoutubeDL instance
+        mock_ydl_instance = MagicMock()
+        
+        # Mock extract_info for download=False and download=True
+        mock_ydl_instance.extract_info.side_effect = [
+            {"title": "Test Video", "ext": "mp4"},  # For download=False
+            {"ext": "mp4"}  # For download=True
+        ]
+        
+        # Set the return value when instantiating YoutubeDL
+        mock_yt_dl_context_manager = MagicMock()
+        mock_yt_dl_context_manager.__enter__.return_value = mock_ydl_instance
+        mock_yt_dlp.return_value = mock_yt_dl_context_manager
+
+        # Test download_media method
+        result = MediaHandler.download_media(self.video_url, self.base_directory)
+        expected_file = Path(self.base_directory) / "Test_Video.mp4"
+        self.assertEqual(result, str(expected_file.resolve()))
+
+    @patch("backend.media_handler.subprocess.run")
+    def test_process_with_media_processor_success(self, mock_subprocess_run):
+        # Mock subprocess.run to simulate a successful response
+        mock_subprocess_run.return_value = MagicMock(
+            returncode=0,
+            stdout="Video processed successfully: /path/to/processed_video.mp4",
+            stderr=""
+            
+        )
+        # Test process_with_media_processor
+        result = MediaHandler.process_with_media_processor(
+            self.video_path, self.base_directory, self.config_path
+        )
+
+        self.assertEqual(result, "/path/to/processed_video.mp4")
```

Comment:
What's happening here with the assertion on l.55?

Reply:
You can see [here](https://github.com/omeryusufyagci/fast-music-remover/blob/32352d65b3a1dc30a3c8fbb1ea01af4241b8012f/MediaProcessor/tests/AudioProcessorTester.cpp#L47) for an example on how a similar test was made for the core. You could reuse the same utility, just need to convert it to python code.

Reply:
Well, `/path/to/...` has to be a real path first. Regarding what we did in the core: we have a known truth, i.e. a processed video that we manually verified by watching it as a reference of what proper filtering is doing to a known source. We then do a byte by byte comparison with that known truth to see if functionality is in tact.

Byte-by-byte is definitely not the best way to do this, but it was a fast solution that actually works with a good tolerance. To be improved, but better than nothing